src/pages/workspaces/workspace/analysis/Analyses.ts

```diff
@@ -543,63 +560,68 @@ const Analyses = _.flow(
         }),
         renamingAnalysisName && h(AnalysisDuplicator, {
           printName: getFileName(renamingAnalysisName),
-          toolLabel: getToolLabelFromFileExtension(renamingAnalysisName),
-          workspaceInfo: { cloudPlatform, googleProject, workspaceId, namespace, name: workspaceName, bucketName },
+          toolLabel: getToolLabelFromFileExtension(getExtension(renamingAnalysisName)),
+          workspaceInfo,
           destroyOld: true,
           fromLauncher: false,
-          onDismiss: () => setRenamingAnalysisName(undefined),
-          onSuccess: () => {
+          onDismiss: async () => {
             setRenamingAnalysisName(undefined)
-            refreshAnalyses()
-          }
+            await refreshAnalyses()
```

Comment:
Does anything await onDismiss or onSuccess? If not, there's no need to wait refreshAnalyses here.

Reply:
Yeah, for the duplicator it is necessary to force the refresh as the modal dismisses for a seamless experience. Will add that to the type signature for the `AnalysisDuplicatorr`

Reply:
Actually, investigating this was a bit tricky. The final solution is that these don't need to be awaited here (which originally was the case because of dependence on `withBusyState` for loading. Now, the component uses the `store` and `LoadedState` pattern to determine busy, and we can fire and forget the `onSuccess` and `onDismiss` for child modals that interact with the data client associated with the store 🎉